### ISSUE TITLE
Infinite loop crashes plugin

### ISSUE THREAD

**Issue by omartan** [lostfields/vscode-nodejs-repl#6]
There must be a way to force exit the plugin in an event where an infinite loop crashes the plugin without having to close the window.

**Comment by lostfields** [comment]
yes, I will try to take a look into this. I may have to use a timeout or code-analysis for an infinite loop to catch this.

Behind the scenes, nodejs repl is used, and infinite loops is a problem here as well.

**Comment by dchicurel** [comment]
Thanks, +1 for this issue. 
Thank you very much for your extension. It is really nice.

I think that trying to analyse if the program will have an infinite loop is an NP problem.. ;) (https://en.wikipedia.org/wiki/Halting_problem)

**Comment by tomprogers** [comment]
I imagine you could establish a timeout like 60 seconds. If the script hasn't completed in that time, assume it's an infinite loop and terminate it. Then, put that number into the extension settings, so folks can change it if 60 seconds is not adequate.

**Comment by LoveSponge** [comment]
Could try to detect a large number of items in the call stack, calling of similar code in quick succession - however you detect, it's worth putting a notification saying something like "infinite loop detected" with actions to 'continue' or 'break' 

EDIT: 
There might be a way to set an event listener for whether the allocated memory has been overdrawn by the running context, see https://devdocs.io/node/v8

**Comment by gplusdotgr** [comment]
@lostfields is there anyway we can help you to do this? it's pretty essential for playing around with loops, algorithms etc (which is I guess one of the main reasons people use node repl)

btw thank you so much for building this!
